### PR TITLE
[MM-55272] Accessibility: Team Icon requires tabbing twice

### PR DESCRIPTION
**Pull request by s-krishnaraju**
#### Summary

-  Only one `Tab` is needed to switch team buttons now 
- Did not add in functionality for `Space` to activate a team button because it is currently used for selecting and swapping the position of team buttons (see GIF below)
- I can add in functionality for `Space` to activate button (i.e same behavior as `Enter`) if that's something we still want to do

#### Ticket Link

GH: https://github.com/mattermost/mattermost/issues/26489
JIRA: https://mattermost.atlassian.net/browse/MM-55272
#### Screenshots

|  Before  |  After  |
|----|----|
|||

| Using `Space` To Switch Icons |
|----|   
||

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@s-krishnaraju could you add plugins such as Github or Gitlab which uses Team sidebar area to see how it effects tabbing with them included

**Review by larkox — APPROVED** [review]

**Comment by s-krishnaraju** [comment]
> @s-krishnaraju could you add plugins such as Github or Gitlab which uses Team sidebar area to see how it effects tabbing with them included

<br>
Tabbing w/ the plugins

File: webapp/channels/src/components/team_sidebar/components/team_button.tsx [hunk 1]
```diff
@@ -192,10 +192,12 @@ export default function TeamButton({
                         ref={provided.innerRef}
                         {...provided.draggableProps}
                         {...provided.dragHandleProps}
+                        tabIndex={-1}
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/team_sidebar/components/team_button.tsx:195]: Do we need two tab indices here?

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: webapp/channels/src/components/team_sidebar/components/team_button.tsx (hunk 1 above)
Comment (s-krishnaraju) [reply webapp/channels/src/components/team_sidebar/components/team_button.tsx:195]: Yeah on closer inspection I don't. On testing, it seems only the draggable-team-container div had an issue with tabbing. I removed the extra tab indices on the team-container divs and it works the same.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
@M-ZubairAhmed 
the tab nagigation between multiple teams on the LHS works.

**Pull request merged**